### PR TITLE
Replace Task.Delay in C# tests

### LINKED ISSUE #3189: csharp/Ice/ami failure (Ubuntu debug)
See https://github.com/zeroc-ice/ice/actions/runs/12014719774/job/33491189118?pr=3187

```
[ running client/server test - 11/25/24 17:11:10 ]
- Config: wss
(dotnet /home/runner/work/ice/ice/csharp/test/Ice/ami/msbuild/server/net8.0/server.dll --Ice.Default.Host=127.0.0.1 --Test.BasePort=14200 --Ice.Warn.Connections=1 --Ice.Default.Protocol=wss --Ice.IPv6=0 --IceSSL.*** --IceSSL.DefaultDir=/home/runner/work/ice/ice/certs --IceSSL.CAs=cacert.pem --IceSSL.VerifyPeer=0 --IceSSL.CertFile=server.p12 --Ice.ThreadPool.Server.Size=1 --Ice.ThreadPool.Server.SizeMax=3 --Ice.ThreadPool.Server.SizeWarn=0 --Ice.PrintAdapterReady=1 --Ice.Trace.Network=2 --Ice.Trace.Retry=1 --Ice.Trace.Protocol=1 --Ice.LogFile=/home/runner/work/ice/ice/csharp/test/Ice/ami/server-112524-1711.log)
(dotnet /home/runner/work/ice/ice/csharp/test/Ice/ami/msbuild/client/net8.0/client.dll --Ice.Default.Host=127.0.0.1 --Test.BasePort=14200 --Ice.Warn.Connections=1 --Ice.Default.Protocol=wss --Ice.IPv6=0 --IceSSL.*** --IceSSL.DefaultDir=/home/runner/work/ice/ice/certs --IceSSL.CAs=cacert.pem --IceSSL.VerifyPeer=0 --IceSSL.CertFile=client.p12 --Ice.Trace.Network=2 --Ice.Trace.Retry=1 --Ice.Trace.Protocol=1 --Ice.LogFile=/home/runner/work/ice/ice/csharp/test/Ice/ami/client-112524-1711.log)
testing async/await...ok
testing local exceptions with async tasks... ok
testing exception with async task... ok
testing progress callback... ok
testing batch requests with proxy and async tasks... ok
testing batch requests with connection... ok
testing batch requests with communicator... ok
testing async/await... System.AggregateException: One or more errors occurred. (Exception of type 'Ice.ConnectionLostException' was thrown.)
 ---> Ice.ConnectionLostException: Exception of type 'Ice.ConnectionLostException' was thrown.
   at Ice.Internal.OutgoingAsyncBase.invokeException() in /home/runner/work/ice/ice/csharp/src/Ice/Internal/OutgoingAsync.cs:line 100
--- End of stack trace from previous location ---
   at Ice.ami.AllTests.<>c__DisplayClass4_0.<<allTestsAsync>b__0>d.MoveNext() in /home/runner/work/ice/ice/csharp/test/Ice/ami/AllTests.cs:line 609
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Ice.ami.AllTests.allTestsAsync(TestHelper helper, Boolean collocated) in /home/runner/work/ice/ice/csharp/test/Ice/ami/AllTests.cs:line 621
   at Ice.ami.Client.runAsync(String[] args) in /home/runner/work/ice/ice/csharp/test/Ice/ami/Client.cs:line 30
   at Test.TestDriver.runTestAsync[T](String[] args) in /home/runner/work/ice/ice/csharp/test/TestCommon/TestHelper.cs:line 232
saved /home/runner/work/ice/ice/csharp/test/Ice/ami/client-112524-1711.log
failed:
   at Test.TestHelper.fail(String message, String detailMessage) in /home/runner/work/ice/ice/csharp/test/TestCommon/TestHelper.cs:line 183
   at Test.TestHelper.test(Boolean b) in /home/runner/work/ice/ice/csharp/test/TestCommon/TestHelper.cs:line 165
   at Ice.ami.TestI.test(Boolean b) in /home/runner/work/ice/ice/csharp/test/Ice/ami/TestI.cs:line 11
   at Ice.ami.TestI.opWithUEAsyncDispatchAsync(Current current) in /home/runner/work/ice/ice/csharp/test/Ice/ami/TestI.cs:line 150
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Ice.ami.TestI.opWithUEAsyncDispatchAsync(Current current)
```

```csharp
            public override async Task
            opWithUEAsyncDispatchAsync(Ice.Current current)
            {
                test(Thread.CurrentThread.Name.Contains("Ice.ThreadPool.Server"));
                await Task.Delay(10);
                test(!Thread.CurrentThread.Name.Contains("Ice.ThreadPool.Server"));  <== line 150
                await self(current).opWithUEAsync();
            }
```

### PR DESCRIPTION
**Pull request by bernardnormier**
Fixes #3189.

### CODE REVIEW

File: csharp/test/Ice/ami/TestI.cs [hunk 1]
```diff
@@ -127,17 +127,13 @@ public override bool
                 return true;
             }
 
-            public override async Task
-            opAsyncDispatchAsync(Ice.Current current)
-            {
-                await Task.Delay(10);
-            }
+            public override Task opAsyncDispatchAsync(Ice.Current current) => Task.CompletedTask;
```
Comment (pepone) [inline csharp/test/Ice/ami/TestI.cs:130]: Shouldn't we use Task.Yield here too, to test asynchronous completion?

File: csharp/test/Ice/operations/MyDerivedClassAMDI.cs [hunk 2]
```diff
@@ -1,829 +1,825 @@
 // Copyright (c) ZeroC, Inc.
 
-namespace Ice
+namespace Ice.operations.AMD;
```
Comment (pepone) [inline csharp/test/Ice/operations/MyDerivedClassAMDI.cs:3]: Is there any fixes in this file other than switching to a file scoped namespace?

**Review by pepone — APPROVED** [review]

File: csharp/test/Ice/ami/TestI.cs (hunk 1 above)
Comment (bernardnormier) [reply csharp/test/Ice/ami/TestI.cs:130]: As far as I can tell, that's not what this test is testing.

File: csharp/test/Ice/operations/MyDerivedClassAMDI.cs (hunk 2 above)
Comment (bernardnormier) [reply csharp/test/Ice/operations/MyDerivedClassAMDI.cs:3]: Yes, removed async and `await Task.Delay(0)`.

**Pull request merged**